fail2ban/tests/fail2banclienttestcase.py

```diff
@@ -0,0 +1,598 @@
+fail2banclient.output = \
+fail2banserver.output = \
+protocol.output = _test_output
+
+def _test_exit(code=0):
+	logSys.debug("Exit with code %s", code)
+	if code == 0:
+		raise ExitException()
+	else:
+		raise FailExitException()   
+fail2bancmdline.exit = \
+fail2banclient.exit = \
+fail2banserver.exit = _test_exit
+
+INTERACT = []
+def _test_input_command(*args):
+	if len(INTERACT):
+		#logSys.debug('--- interact command: %r', INTERACT[0])
+		return INTERACT.pop(0)
+	else:
+		return "exit" 
+fail2banclient.input_command = _test_input_command
+
+# prevents change logging params, log capturing, etc:
+fail2bancmdline.PRODUCTION = \
+fail2banclient.PRODUCTION = \
+fail2banserver.PRODUCTION = False
+
+
+class ExitException(fail2bancmdline.ExitException):
+	pass
+class FailExitException(fail2bancmdline.ExitException):
+	pass
+
+
+def _out_file(fn): # pragma: no cover
+	logSys.debug('---- ' + fn + ' ----')
+	for line in fileinput.input(fn):
+		line = line.rstrip('\n')
+		logSys.debug(line)
+	logSys.debug('-'*30)
+
+def _start_params(tmp, use_stock=False, logtarget="/dev/null"):
+	cfg = tmp+"/config"
+	if use_stock and STOCK:
+		# copy config (sub-directories as alias):
+		def ig_dirs(dir, files):
+			return [f for f in files if os.path.isdir(os.path.join(dir, f))]
+		shutil.copytree(STOCK_CONF_DIR, cfg, ignore=ig_dirs)
+		os.symlink(STOCK_CONF_DIR+"/action.d", cfg+"/action.d")
+		os.symlink(STOCK_CONF_DIR+"/filter.d", cfg+"/filter.d")
+		# replace fail2ban params (database with memory):
+		r = re.compile(r'^dbfile\s*=')
+		for line in fileinput.input(cfg+"/fail2ban.conf", inplace=True):
+			line = line.rstrip('\n')
+			if r.match(line):
+				line = "dbfile = :memory:"
+			print(line)
+		# replace jail params (polling as backend to be fast in initialize):
+		r = re.compile(r'^backend\s*=')
+		for line in fileinput.input(cfg+"/jail.conf", inplace=True):
+			line = line.rstrip('\n')
+			if r.match(line):
+				line = "backend = polling"
+			print(line)
+	else:
+		# just empty config directory without anything (only fail2ban.conf/jail.conf):
+		os.mkdir(cfg)
+		f = open(cfg+"/fail2ban.conf", "w")
+		f.write('\n'.join((
+			"[Definition]",
+			"loglevel = INFO",
+			"logtarget = " + logtarget,
+			"syslogsocket = auto",
+			"socket = "+tmp+"/f2b.sock",
+			"pidfile = "+tmp+"/f2b.pid",
+			"backend = polling",
+			"dbfile = :memory:",
+			"dbpurgeage = 1d",
+			"",
+		)))
+		f.close()
+		f = open(cfg+"/jail.conf", "w")
+		f.write('\n'.join((
+			"[INCLUDES]", "",
+			"[DEFAULT]", "",
+			"",
+		)))
+		f.close()
+		if logSys.level < logging.DEBUG: # if HEAVYDEBUG
+			_out_file(cfg+"/fail2ban.conf")
+			_out_file(cfg+"/jail.conf")
+	# parameters (sock/pid and config, increase verbosity, set log, etc.):
+	return ("-c", cfg, "-s", tmp+"/f2b.sock", "-p", tmp+"/f2b.pid",
+					"-vv", "--logtarget", logtarget, "--loglevel", "DEBUG", "--syslogsocket", "auto",
+					"--timeout", str(fail2bancmdline.MAX_WAITTIME),
+	)
+
+def _kill_srv(pidfile): # pragma: no cover
```

Comment:
This is only test case helper, used in helper decorator with_kill_srv only for safe removing server if some test case failed (remove process, free socket, etc.). 
So imho at all not expected to be covered - even test case helper only.